x-pack/plugin/esql/qa/testFixtures/src/main/resources/enrich.csv-spec

```diff
@@ -650,6 +650,7 @@ IDR       |  Indore       |  POINT(75.8472 22.7167)  |  India      |  POINT(75.8
 fieldsInOtherIndicesBug
 required_capability: enrich_load
 required_capability: fix_replace_missing_field_with_null_duplicate_name_id_in_layout
+required_capability: dense_vector_field_type
```

Comment:
This and below addition of `required_capability: dense_vector_field_type`.

Previously enrich resolutions were happening before main field caps in order to resolve `matchField` that is later added to the list of fields in the main field caps call.

In order to make main field caps call before enrich we have to request all fields in case there is any enrich in the query (as we do not know what might be their matchField yet). This list is kept and serialized within the plan. For this two affected queries we happen to query `from *` that includes a `dense_vector` with corresponding `dense_vector` field. It is not supported prior to 9.2. I am adding this capability in order to be able to deserialize a plan with this field on data nodes.